Merge Entities branch on Master

jbaptisteq opened the pull request:
All feats add.

To Do after merge : 
- Minor change graphic interface
- Update docs
- Final Fixtures

jbaptisteq committed:
feat (<entity>) : Create entities and Bundle
- Create category for CoreBundle
- Create img for CoreBundle
- Create vid for CoreBundle

- Create UserBundle with default files
- Create user for UserBundle
- Create avatar for UserBundle
- Create comment for UserBundle

jbaptisteq committed:
feat (<entity>) : Create entities and Bundle
- Create #8 category for CoreBundle
- Create #9 vid for CoreBundle
- Create #10 img for CoreBundle

- Create #14 UserBundle with default files
- Create #11 avatar for UserBundle
- Create #12 user for UserBundle
- Create #13 comment for UserBundle

jbaptisteq committed:
Merge branch 'Entities_Branch' of https://github.com/jbaptisteq/OC_P6 into Entities_Branch

jbaptisteq committed:
fix : Improvement for trick list.
- fix problem with sizing in css
- fix routing problem with UserBundle

jbaptisteq committed:
feat (<entity>) : Create Category
- Update category and trick entity for implement relation.
- Update services.yml for include Category fixture.
- Update trick fixture
- Upload category fixture

jbaptisteq committed:
feat (<mobile>) : Create Bookmarl link
- Upload bookmark.js
- Update main.css

jbaptisteq committed:
feat (<user management>)
- Create Login and Logout system
- Create Register system
- Update View with layout, login, register form
- Update user entity for some conflit with relation between table
- Clean code with removing some useless files

jbaptisteq committed:
build (<symfony>) Update
- Version 3.4.14 from 3.4.13

jbaptisteq committed:
fix (<symfony>)
- Add description in composer.json

jbaptisteq committed:
fix (<register>) Fix register error
- Remove plaintext password

jbaptisteq committed:
style : minox fix in psr

jbaptisteq committed:
feat (<forgot password>)
- routing for form
- forgot password form
- forget function in security controller
- add link in login template

jbaptisteq committed:
refactor : Change Class names

jbaptisteq committed:
feat (<account> : email validation
- add view for email validation by token
- update routing.yml, RegistrationController.php, main.css, login View,
Register view, entity\User.php

jbaptisteq committed:
divers : Adding working files

jbaptisteq committed:
divers : update working files

jbaptisteq committed:
fix(<account>) : remove useless comment

jbaptisteq committed:
feat(<login>) : Log only validate account
- Update user.php
- Update Login view
- Update RegistrationController

jbaptisteq committed:
fix : Rename Variable in User.php

jbaptisteq committed:
feat (<account>) : Password change
- Update view and file for adding change password feat.
- Upload new model for mail, change password feat.

jbaptisteq committed:
refactor : SecurityController.php

jbaptisteq committed:
refactor : UserBundle Controller

jbaptisteq committed:
style : delete comments

jbaptisteq committed:
style : Correct Comment
- Delete useless file

jbaptisteq committed:
refactor : Entity
Img -> Image
Vid -> Video

jbaptisteq committed:
divers : upload asset files for trick View

jbaptisteq committed:
feat : Trick View with fixture

jbaptisteq committed:
divers : update symfony 3.4.15

jbaptisteq committed:
feat (<route>) : Add route for all function in future feat.

jbaptisteq committed:
feat (<trick>) :
- Addind unique constraint on name
- Adding published var for request on tricks in index view
- Adding datetime management for creation and edition

jbaptisteq committed:
feat (<trick>)
- adding image management, upload, check and name modification.

jbaptisteq committed:
divers : update service

jbaptisteq committed:
refactor : avatar entity

jbaptisteq committed:
feat (<comment>)
update comment entity with relationORM to trick and user.

jbaptisteq committed:
feat (<trick>) : adding new trick
- Forms
- View
- js script

jbaptisteq committed:
feat (<trick>)
-script for media button on mobile view

jbaptisteq committed:
feat (<trick>) Management
- view for edit and delete trick.

jbaptisteq committed:
feat (<trick>) : upload controller with functions
- add trick
- edit trick
- delete trick
- delete image on trick view
- delete video on trick view
- update image on trick view
- update video on trick view
- show more comment

jbaptisteq committed:
feat :
- update view for index, trick and layout.
- style : trick repository

jbaptisteq committed:
feat : update Indexcontroller and trick controller
- management index view for 15 tricks and show more
- management trick view for comment
- add comment function
- correct nbcomment on triclcontroller

jbaptisteq committed:
style : update css files

jbaptisteq committed:
feat (<trick>) adding script for media management

jbaptisteq committed:
asset : upload default image for avatar

Fschardt commented on app/config/config_dev.yml:
```diff
@@ -38,4 +38,6 @@ monolog:
         #    level: info
 
 #swiftmailer:
```
Can be deleted if not used. No need to keep it commented

Fschardt commented on app/config/routing.yml:
```diff
@@ -1,3 +1,42 @@
     prefix:   /
+
+homepage:
+    path:    /
```
Only one space after path:

jbaptisteq replied:
Code corrected

Fschardt commented on app/config/routing.yml:
```diff
@@ -1,3 +1,42 @@
+    defaults: { _controller: p6CoreBundle:Index:index }
+
+register:
+    path:     /register
```
Only one space after path:

jbaptisteq replied:
Code corrected

Fschardt commented on app/config/security.yml:
```diff
@@ -2,10 +2,28 @@
     providers:
-        in_memory:
-            memory: ~
+        #in_memory:
```
Can be deleted. No need to keep it commented

jbaptisteq replied:
Code delete

Fschardt commented on src/p6/CoreBundle/Controller/IndexController.php:
```diff
@@ -6,21 +6,38 @@
+      $entityManager = $this->getDoctrine()->getManager();
+      $trick = $entityManager->getRepository('p6CoreBundle:Trick')->find($id);
+
+      if(null === $trick){
```
PSR2 : if (null === $trick) {

jbaptisteq replied:
PSR2 apply

Fschardt commented on src/p6/CoreBundle/Resources/views/Index/index.html.twig:
```diff
@@ -3,31 +3,47 @@
 <div class="container desktop">
     <div class="row">
-{% set nbtrick = 5 %}
+{% set nbtrick = 15 %}
```
So, what if we have more than 15 tricks to display ?

jbaptisteq replied:
line doesnt exist anymore. Replace by function.

Fschardt commented on src/p6/UserBundle/Controller/RegistrationController.php:
```diff
@@ -0,0 +1,100 @@
+            $entityManager->flush();
+
+            return $this->redirectToRoute('homepage');
+            }
```
Rerun indentation on whole file. From line 67 there is one indentation level that should not exist

Fschardt commented on src/p6/UserBundle/Controller/SecurityController.php:
```diff
@@ -0,0 +1,94 @@
+  */
+  public function loginAction()
+  {
+    if ($this->get('security.authorization_checker')->isGranted('IS_AUTHENTICATED_REMEMBERED')){
```
PSR2 : space before {

jbaptisteq replied:
Corrected

Fschardt commented on src/p6/UserBundle/Resources/config/services.yml:
```diff
@@ -0,0 +1,12 @@
+services:
+#    p6_user.example:
```
To remove

jbaptisteq replied:
clear

Fschardt reviewed (changes requested):
De manière générale, ne pas laisser de code commenté inutile, attention également aux lignes vides

jbaptisteq committed:
Review :
- All corrected from PR #36

jbaptisteq committed:
Fix : Issue #34
- Modify trick view for image display
- Modify index view, adding Up arrow over 15 tricks display.
- Fix input in mobile view for edit and delete media

jbaptisteq committed:
feat : Fixture for tricks

jbaptisteq committed:
docs : issue #35
- Update Diagrams
- Update Readme.md
- Upload ocp6_jbq.sql
- Adding an asset for this database version

jbaptisteq committed:
refactor : lisibility

Fschardt commented on src/p6/CoreBundle/Controller/TrickController.php:
```diff
@@ -0,0 +1,252 @@
+    $trick = $entityManager->getRepository('p6CoreBundle:Trick')->find($id);
+    $form = $this->createForm(TrickType::class, $trick);
+
+    if(null === $trick){
```
PSR2

Fschardt commented on src/p6/CoreBundle/Controller/TrickController.php:
```diff
@@ -0,0 +1,252 @@
+    $trick = $entityManager->getRepository('p6CoreBundle:Trick')->find($id);
+    $form = $this->createForm(TrickType::class, $trick);
+
+    if(null === $trick){
```
PSR2

jbaptisteq replied:
Done

Fschardt commented on src/p6/CoreBundle/Controller/TrickController.php:
```diff
@@ -0,0 +1,252 @@
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de la suppréssion de l\'image.');
```
Can be deleted with 143

jbaptisteq replied:
Done

Fschardt commented on src/p6/CoreBundle/Controller/TrickController.php:
```diff
@@ -0,0 +1,252 @@
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de la suppréssion de la vidéo');
```
Can be deleted with 172

jbaptisteq replied:
Done

Fschardt commented on src/p6/CoreBundle/Controller/TrickController.php:
```diff
@@ -0,0 +1,252 @@
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de l\'édition de l\'image.');
```
Can be deleted with 203

jbaptisteq replied:
Done

Fschardt commented on src/p6/CoreBundle/Controller/TrickController.php:
```diff
@@ -0,0 +1,252 @@
+
+        return $this->redirectToRoute('trick', array('id' => $idtrick));
+      }
+      $this->addFlash('alert', 'Erreur lors de la modification de la vidéo');
```
Can be deleted with 235

jbaptisteq replied:
Done

jbaptisteq committed:
Refactor : TrickController.php
- Correction useless Code
- Correction PSR-2 on If

jbaptisteq commented:
Correction  and Commit  done.

Le mar. 18 sept. 2018 à 16:59, Fschardt <notifications@github.com> a écrit :

> *@Fschardt* requested changes on this pull request.
> ------------------------------
>
> In src/p6/CoreBundle/Controller/TrickController.php
> <https://github.com/jbaptisteq/OC_P6/pull/36#discussion_r218464461>:
>
> > +      return $this->redirectToRoute('homepage');
> +    }
> +
> +    return $this->render('p6CoreBundle:Trick:add.html.twig', array('form' => $form->createView(),));
> +  }
> +
> +  /**
> +  * @Route("/editTrick/{id}", name="editTrick")
> +  */
> +  public function editAction(Request $request, $id)
> +  {
> +    $entityManager = $this->getDoctrine()->getManager();
> +    $trick = $entityManager->getRepository('p6CoreBundle:Trick')->find($id);
> +    $form = $this->createForm(TrickType::class, $trick);
> +
> +    if(null === $trick){
>
> PSR2
> ------------------------------
>
> In src/p6/CoreBundle/Controller/TrickController.php
> <https://github.com/jbaptisteq/OC_P6/pull/36#discussion_r218465496>:
>
> > +
> +    if ($request->isMethod('POST')) {
> +
> +      $submittedToken = $request->request->get('token');
> +      if ($this->isCsrfTokenValid('delete-item', $submittedToken)) {
> +
> +        $entityManager = $this->getDoctrine()->getManager();
> +        $image = $entityManager->getRepository('p6CoreBundle:Image')->find($id);
> +        $entityManager->remove($image);
> +        $entityManager->flush();
> +
> +        $this->addFlash('success', 'Image Supprimée.');
> +
> +        return $this->redirectToRoute('trick', array('id' => $idtrick));
> +      }
> +      $this->addFlash('alert', 'Erreur lors de la suppréssion de l\'image.');
>
> Can be deleted with 143
> ------------------------------
>
> In src/p6/CoreBundle/Controller/TrickController.php
> <https://github.com/jbaptisteq/OC_P6/pull/36#discussion_r218466079>:
>
> > +
> +    if ($request->isMethod('POST')) {
> +
> +      $submittedToken = $request->request->get('token');
> +      if ($this->isCsrfTokenValid('delete-item', $submittedToken)) {
> +
> +        $entityManager = $this->getDoctrine()->getManager();
> +        $video = $entityManager->getRepository('p6CoreBundle:Video')->find($id);
> +        $entityManager->remove($video);
> +        $entityManager->flush();
> +
> +        $this->addFlash('success', 'Vidéo Supprimée.');
> +
> +        return $this->redirectToRoute('trick', array('id' => $idtrick));
> +      }
> +      $this->addFlash('alert', 'Erreur lors de la suppréssion de la vidéo');
>
> Can be deleted with 172
> ------------------------------
>
> In src/p6/CoreBundle/Controller/TrickController.php
> <https://github.com/jbaptisteq/OC_P6/pull/36#discussion_r218466248>:
>
> > +
> +      $submittedToken = $request->request->get('token');
> +      if ($this->isCsrfTokenValid('update-item', $submittedToken)) {
> +
> +        $entityManager = $this->getDoctrine()->getManager();
> +        $image = $entityManager->getRepository('p6CoreBundle:Image')->find($id);
> +
> +        $newImage = $request->request->get('newImage');
> +        $image->setUrl($newImage);
> +        $entityManager->flush();
> +
> +        $this->addFlash('success', 'Image Editée.');
> +
> +        return $this->redirectToRoute('trick', array('id' => $idtrick));
> +      }
> +      $this->addFlash('alert', 'Erreur lors de l\'édition de l\'image.');
>
> Can be deleted with 203
> ------------------------------
>
> In src/p6/CoreBundle/Controller/TrickController.php
> <https://github.com/jbaptisteq/OC_P6/pull/36#discussion_r218466382>:
>
> > +      $submittedToken = $request->request->get('token');
> +      if ($this->isCsrfTokenValid('update-item', $submittedToken)) {
> +
> +        $entityManager = $this->getDoctrine()->getManager();
> +        $video = $entityManager->getRepository('p6CoreBundle:Video')->find($id);
> +
> +        $newLink = $request->request->get('newLink');
> +        $video->setLink($newLink);
> +
> +        $entityManager->flush();
> +
> +        $this->addFlash('success', 'Vidéo Modifiée');
> +
> +        return $this->redirectToRoute('trick', array('id' => $idtrick));
> +      }
> +      $this->addFlash('alert', 'Erreur lors de la modification de la vidéo');
>
> Can be deleted with 235
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jbaptisteq/OC_P6/pull/36#pullrequestreview-156408403>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AgaFNPSoO4wEp5ivuitUBTjd_pC1Pa8kks5ucQpAgaJpZM4WlFRp>
> .
>

jbaptisteq committed:
Refactor : indexController.php
